crates/static-file/static-file/src/segments/receipts.rs

```diff
@@ -42,11 +42,8 @@ impl<DB: Database> Segment<DB> for Receipts {
             let mut receipts_cursor = provider.tx_ref().cursor_read::<tables::Receipts>()?;
             let receipts_walker = receipts_cursor.walk_range(block_body_indices.tx_num_range())?;
 
-            for entry in receipts_walker {
-                let (tx_number, receipt) = entry?;
-
-                static_file_writer.append_receipt(tx_number, receipt)?;
-            }
+            let receipts = receipts_walker.collect::<Result<Vec<_>, _>>()?;
+            static_file_writer.append_receipts(receipts.into_iter().map(Ok))?;
```

Comment:
that should work now, and the point of accepting the iterator of `Result` is to not collect the receipts into a vector here

```suggestion
            static_file_writer.append_receipts(receipts_walker)?;
```